File: sources/scala/07_string_append/main.scala
// task 07 string_append — expected output: 1000000
// build: scalac -release 17 main.scala    run: scala Main

object Main {
  def main(args: Array[String]): Unit = {
    var text = ""
    var i = 0L
    while (i < 1000000L) {
      text = text + "x"
      i += 1L
    }
    println(text.length)
  }
}

File: sources/nim/07_string_append/main.nim
# task 07 string_append — expected output: 1000000
# build: nim c -d:release -o:prog main.nim    run: ./prog
#
# `&` builds a new string, so every append copies the whole string (quadratic),
# which is what this task measures. `add` would grow the buffer in place instead.

var text = ""
for _ in 0 ..< 1_000_000:
  text = text & "x"
echo text.len
